packages/core/src/components/Table/Table.tsx

```diff
@@ -0,0 +1,177 @@
+/*
+ * Copyright 2020 Spotify AB
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { makeStyles } from '@material-ui/core';
+import { BackstageTheme } from '@backstage/theme';
+import React, { FC, useCallback } from 'react';
+import { useTable, useGlobalFilter, useSortBy } from 'react-table';
+
+import { FixedSizeList } from 'react-window';
+import MUITable from '@material-ui/core/Table';
+import TableBody from '@material-ui/core/TableBody';
+import TableCell from '@material-ui/core/TableCell';
+import TableContainer from '@material-ui/core/TableContainer';
+import TableHead from '@material-ui/core/TableHead';
+import TableRow from '@material-ui/core/TableRow';
+import TableSortLabel from '@material-ui/core/TableSortLabel';
+
+import TextField from '@material-ui/core/TextField';
+
+//const useStyles = makeStyles<typeof BackstageTheme>(theme => ({ }));
+type GlobalFilterProps = {
+  setGlobalFilter: any;
+  preGlobalFilteredRows: any;
+  globalFilter: any;
+};
+
+const GlobalFilter: FC<GlobalFilterProps> = ({
+  setGlobalFilter,
+  preGlobalFilteredRows,
+  globalFilter,
+}) => {
+  const count = preGlobalFilteredRows.length;
+
+  return (
+    <span>
+      <TextField
+        value={globalFilter || ''}
+        placeholder={`${count} records...`}
+        id="standard-basic"
+        label="Search"
+        onChange={e => {
+          setGlobalFilter(e.target.value || undefined); // Set undefined to remove the filter entirely
+        }}
+      />
+    </span>
+  );
+};
+
+type TableProps = {
+  data: Array<any>;
+  columns: Array<any>;
```

Comment:
A toggle for showing/hiding the search bar is relevant I think.

Reply:
In general I'm wondering if we really want to create out own API here. Internally I think we solve 80% of the usages by just copying stuff, but the docs have been important for the rest imo. Compare to the docs at https://material-table.com/ for example, not sure we want to replicate that.

The strategy could be that we provide this table component to solve 80%, and then let people use `react-table` directly for the rest.

Reply:
@Rugvip my take is that we should have a super simple table component that covers 60-80%. Our internal version has grown to cover ~95% which also meant more complexity and poor docs.

One way to make this clear is to have 2 different components in our Storybook. One for the simple use-cases (this one) and a more powerful (Complex Table) version that uses Material Table as is.

Reply:
I was looking at material-table, but could not find any support for virtualization. If that is a requirement at some point we might have to find a third table as well. That why i went with react-table for this example as it allows us to start small and reasonably easy add the features we want without adding stuff we don't need.
Other than that material-table looks really nice though, so if i'm mistaken about it supporting virtualization or if we know we won't need it i'm all for it. :)

Reply:
I agree with having a prop to hide the search. A great feature set for a basic table would be sorting by clicking headers, and filtering that just works. Having both by default and then have props for (1) disabling sorting, or (2) hiding headers entirely, and (3) hiding the filter box, would go a long way toward a great but easy to use table for that 80% case.